server/src/main/java/io/seata/server/session/SessionHelper.java

```diff
@@ -201,6 +205,10 @@ public static void endRollbacked(GlobalSession globalSession, boolean retryGloba
             MetricsPublisher.postSessionDoneEvent(globalSession, IdConstants.STATUS_VALUE_AFTER_ROLLBACKED_KEY, true,
                     beginTime, retryBranch);
         } else {
+            if (globalSession.isSaga()) {
```

Comment:
LGTM.  This PR satisfies the functional requirements, but the common method to future designs should be decoupled from the transaction mode.